### ISSUE TITLE
wrong characters

### ISSUE THREAD

**Issue by simonedd** [UglyToad/PdfPig#286]
Hi,

When I read the attached files I get some strange wrong characters, like DĂĕŽŶŶĞƌŝĞĞŶĚƵŝƚĞ.
I tested it with the AdvancedTextExtraction sample.
[Facade.pdf](https://github.com/UglyToad/PdfPig/files/5956671/Facade.pdf)

Thanks,
Simone

**Comment by Poltuu** [comment]
As far as I could tell, it seems like those words _are_ in the pdf somehow. I would suggest trying another library and compare to confirm that such text appears or not. The text could be hidden behind the images for instance.

**Comment by simonedd** [comment]
You're right, I have the same wrong characters with another library, tested with iText 7.
But I don't see any images in the file and opening it with Adobe the text looks like text. Could there be something else covering the word?
In the following pictures, you can see an example of a word that looks not correct.
![image](https://user-images.githubusercontent.com/31410227/107505964-b378ad80-6b9d-11eb-9a39-d92e16b6c0f4.png)
![image](https://user-images.githubusercontent.com/31410227/107506038-d014e580-6b9d-11eb-83f6-1875bd7df52c.png)

**Comment by BobLd** [comment]
Out of curiosity, how did you render the second image? And what are these red boxes? (I guess the first one is a screenshot from Acrobat reader)

**Comment by simonedd** [comment]
I'm rendering it with SharpDX. Red boxes are the GlyphRectangle of the first letter of each word, as you can see I have some problem with text size, is a little bigger.

**Comment by EliotJones** [comment]
Copying text from the PDF inside both Firefox and Edge (Chromium) copies these letters in both browsers which suggests they are genuinely in the content of the document. It may be the case that they are overlaid by the rectangle boxes of the annotations.

This flags a potential need to tell which paths appear above/below each other and glyphs.

**Comment by Poltuu** [comment]
It might be totally wrong, but I also noticed the presence of "optional content" on the pdf, and I wondered if this had to do with it. The optio al content itself is unrelated to the issue (other correct text), but I had never crossed that before.

**Comment by simonedd** [comment]
Yes, copying from the browser I get these characters, but I don't get any annotation with Page.ExperimentalAccess.GetAnnotations()

**Comment by simonedd** [comment]
Hi,

I have another file with the same issue of wrong characters, e.g. "ĭZ\P:". I don't see any annotations that cover the text so I guess there is something wrong with the characters decoding.
[JWO_IS_eksport_pdf_..pdf](https://github.com/UglyToad/PdfPig/files/6960226/JWO_IS_eksport_pdf_.pdf)

**Comment by dkiltyhc** [comment]
This also applies to form fields. It seems to be an issue with extended character sets. For example in one of the fields

"Formulaire de déclaration obligatoire des incidents"
accented or apostrophe are rendered strangely. When I output the characters to json in my app, somehow they do get escaped to unicode  like this: "Formulaire de d\u00E9claration obligatoire des incidents"



[md-mm_form-fra-test1.pdf](https://github.com/UglyToad/PdfPig/files/8624607/md-mm_form-fra-test1.pdf)
s

**Comment by BobLd** [comment]
Closing as the font chars issues are covered in more recent issues.

**Comment by simonedd** [comment]
Could you link the related issues so I can keep track of any improvements on chars parsing? Thanks
